### PR TITLE
Upgrade Navigation library

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
upgrade react-native-navigation to version 6.0.1 and use Android auto linking support
#### Ticket Link
TBD

Test all screens on both Android and iOS in portrait and landscape

### CODE REVIEW

**Comment by enahum** [comment]
@matthewbirtch PR updated. As for the height of the navbar in landscape, I still see it OK. Just out of curiosity what device do you have?

**Comment by matthewbirtch** [comment]
@enahum I’m on an iPhone 11 with iOS 13.3.1

**Comment by enahum** [comment]
@josephbaylon you are right, but not sure if we can do anything about it.. that code did not change, only the underlying dependency.. if it is cause of the dependency then I guess there is not much we can do.

What I see is the navbar changing color before the rest of the screen. Indeed I would agree is not perfect but it is not horrible IMHO and I'm not sure based on the likelihood if we should spend time hacking the library to make it smoother. thoughts?

**Comment by matthewbirtch** [comment]
@enahum I installed the latest build for 3997 from the QA channel and I'm still seeing the last issue I commented on. I think everything else is ok though.

File: android/app/src/main/java/com/mattermost/rnbeta/MainApplication.java [hunk 1]
```diff
@@ -87,78 +54,34 @@
...
-    // No need to add RnnPackage and MainReactPackage
-    return Arrays.<ReactPackage>asList(
-            new TurboReactPackage() {
+    @Override
+    protected List<ReactPackage> getPackages() {
+      @SuppressWarnings("UnnecessaryLocalVariable")
+      List<ReactPackage> packages = new PackageList(this).getPackages();
+      // Packages that cannot be autolinked yet can be added manually here, for example:
+      // packages.add(new MyReactNativePackage());
+      packages.add(new RNNotificationsPackage(MainApplication.this));
```
Comment (chuttam) [inline android/app/src/main/java/com/mattermost/rnbeta/MainApplication.java]: Would passing `instance` be equivalent?

Comment (enahum) [reply android/app/src/main/java/com/mattermost/rnbeta/MainApplication.java]: I guess it would, but I don't want to deviate from the docs

**Pull request merged**